sklearn/tree/_utils.pyx

```diff
@@ -518,20 +521,22 @@ cdef class WeightedMedianCalculator:
         Return -1 in case of failure to allocate memory (and raise MemoryError)
         or 0 otherwise.
         """
-        cdef int return_value
+        cdef int push_index
         cdef DOUBLE_t original_median
 
         if self.size() != 0:
             original_median = self.get_median()
         # samples.push (WeightedPQueue.push) uses safe_realloc, hence except -1
-        return_value = self.samples.push(data, weight)
-        self.update_median_parameters_post_push(data, weight,
-                                                original_median)
-        return return_value
+        push_index = self.samples.push(data, weight)
+        if push_index == -1:
```

Comment:
>if push_index == -1:
>    return -1

I added this to replicate what was previously being returned.  Though we should only get a -1 when an exception occurs i.e., MemoryError.  **So in hindsight I think I can remove this check** since an exception in self.samples.push should terminate immediately...?  I am not 100% sure though with being new to Cyphon?